Add Nix flake

nilp0inter opened the pull request:
- This PR adds a flake with a derivation using the current nixpkgs's Nim infrastructure.
- The previous nix code is unchanged.
- A small documentation section is added to the README.

nilp0inter committed:
Add Nix flake

Lord-Valen commented on flake.nix:
```diff
@@ -0,0 +1,57 @@
+                libGL
+                libGLU
+              ];
+            };
```
No `meta.licence` or `meta.mainProgram`?

Lord-Valen replied:
> `meta.mainProgram` is not specified when the binary name matches the derivation name

That was because tools would assume that the main program is the same as the pname. Now, that is only the case for `nix run`, `lib.getExe` used to make that assumption but it's a bad assumption and was deprecated. https://github.com/NixOS/nixpkgs/pull/246386

nilp0inter replied:
Thanks for the reference! I've added `mainProgram`

nilp0inter commented:
Ping @Lord-Valen

leana8959 commented:
Hello,

I have tried to use the executable built using the nix flake proposed by this PR, however it gives an error "No appropriate visual found", is this related to a bad build process, or am I using the program incorrectly?

```
[~] $ result/bin/boomer
/home/leana/.config/boomer/config doesn't exist. Using default values.
Using config: (min_scale: 0.01, scroll_speed: 1.5, drag_friction: 6.0, scale_friction: 4.0)
Screen rate: 60
GLX version 1.4
GLX extension:
No appropriate visual found
```

Thanks :)

nilp0inter commented:
@leana8959,

During this year, nixpkgs's Nim machinery changed quite a bit. I updated and simplified this PR, taking advantage of the new function `buildNimPackage` and now it should work just fine.

Try running:

```console
nix run github:nilp0inter/boomer
```

Cheers,
Roberto

leana8959 commented:
I confirm that this version of the flake does work, thank you!

Have a nice day :)
